### ISSUE TITLE
Chunks not stored

### ISSUE THREAD

**Issue by davidbrochart** [davidbrochart/pangeo-forge-recipes#1]
I [changed the recipe](https://github.com/davidbrochart/pangeo-forge-recipes/commit/d789a43efb80fb69a801a71f2a8f5552e4bdb1d4) so that it is executed in an executor. But [in the logs](https://github.com/davidbrochart/pangeo-forge-recipes/runs/1797228169) we can see that no chunk is written.
Am I missing something @rabernat?

**Comment by rabernat** [comment]
```
OSError: [Errno 28] No space left on device
```

Looks like you are filling up the disk on the github workflows VM.

It was never the intention that a full recipe would be runnable inside a github workflow. That's what we need the bakeries for.

**Comment by davidbrochart** [comment]
Yes, I was expecting to run out of disk space, and I am just playing with the recipe here, but what I don't understand is that it only caches input data, and never write chunks in the target.

**Comment by rabernat** [comment]
My impression is that it fails on the cache step so it can't move forward to the later steps. That's definitely the behavior I would have expected here.

**Comment by davidbrochart** [comment]
I reduced the input file list and chunk size to a minimum (2 chunks of 2 inputs each), and indeed the recipe produces an output. I'm surprised by the amount of caching that was made previously, it ran out of space caching 2000 input files, which corresponded to 500 chunks, and it hadn't started producing any output yet.
Is there a way to control the amount of caching in the executor?

**Comment by rabernat** [comment]
The current design is that *all* the inputs will be cached before moving on to storing chunks.

**Comment by davidbrochart** [comment]
Oh, that's not what I had imagined, thanks for the clarification.
So would this recipe be ready? When I look at the [example recipe](https://github.com/pangeo-forge/staged-recipes/blob/master/recipes/example/pipeline.py), I can see that it's organized very differently. If I used a `PrefectPipelineExecutor`, would they be equivalent anyway?

**Comment by rabernat** [comment]
The example is totally stale now that we have refactored pangeo forge.

You have raced ahead and are now in uncharted territory! 😉
